O/ORTools/build_tarballs.jl

```diff
@@ -0,0 +1,54 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "ORTools"
+version = v"9.7"
+
+# Collection of sources required to build this package
+sources = [
+    GitSource("https://github.com/google/or-tools.git",
+              "35d56a4b07db8ea135b83762289fbc0e7d229221"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/or-tools*
+mkdir build
+cmake -S. -Bbuild \
```

Comment:
https://docs.binarybuilder.org/stable/build_tips/#CMake-builds You have to pass `-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}` otherwise you're building for the wrong platform.

Reply:
Thanks! I didn't find that section on that page (as I'm not on Linux — or without Docker/VM access when on Linux —, I couldn't use the wizard).

Reply:
Actually, using this option makes things worse: it messes up with paths in such a way that `${CMAKE_CURRENT_LIST_DIR}` isn't pointing to the expected thing, so that the mechanism inside OR-Tools to apply patches fails (https://github.com/google/or-tools/blob/1d696f9108a0ebfd99feb73b9211e2f5a6b0812b/cmake/host.CMakeLists.txt#L106).

Reply:
Sounds like an upstream bug then, the toolchain file ***has*** to be used.

Reply:
> Actually, using this option makes things worse: it messes up with paths in such a way that ${CMAKE_CURRENT_LIST_DIR} isn't pointing to the expected thing

It's pointing to the correct place - binary builder cross compiles for all architectures, so the or-tools build chain gets split between a host section (executables used during the build process) and the or-tools section. The error is happening in the host section. In cross compilation, it tries to compile the host tools in the `build/host_tools` directory using a CMakeLists.txt file rooted there. That means all the patches for the host code are only 2 levels down from the root repository instead of 3 levels, so that is a bug with or-tools.

My recommendation is that or-tools not try fussing with relative paths at all, and instead pass the correct paths into their cmake build system for the host tools that they configure. e.g., I could see the following working best:
* Modify the CMake command that generates the host tools to pass an additional variable `-DORTOOLS_GIT_ROOT=${CMAKE_CURRENT_SOURCE_DIR}` here https://github.com/google/or-tools/blob/1d696f9108a0ebfd99feb73b9211e2f5a6b0812b/cmake/host.cmake#L53.
* They then modify the three patch lines (https://github.com/google/or-tools/blob/1d696f9108a0ebfd99feb73b9211e2f5a6b0812b/cmake/host.CMakeLists.txt#L106, https://github.com/google/or-tools/blob/1d696f9108a0ebfd99feb73b9211e2f5a6b0812b/cmake/host.CMakeLists.txt#L120 and https://github.com/google/or-tools/blob/1d696f9108a0ebfd99feb73b9211e2f5a6b0812b/cmake/host.CMakeLists.txt#L136) to instead be `${ORTOOLS_GIT_ROOT}/patches/<patch_name>` instead.

Reply:
Thanks for stopping by! I had hoped that our CMake infrastructure was ready for cross-compilation, but it clearly is not. Based on the errors I got at other places, it's not just a question of relative paths (looks like `protoc` is compiled for ARM64 while run on x86_64 for Linux ARM64, for instance). Fixing that small issue will only work in one case, but a truckload of others seem to wait just after…

Given how understaffed we are regarding open-source support in general and build systems in particular, the only feasible solution I see in the short term is to forego cross-compilation and focus on Linux x64 (using our existing prebuilt binaries for the other platforms). It would be best if we could support cross-compilation, but this is the first time it's being required for something. (And we don't have a huge buy-in regarding Julia support, making it harder to justify such a time investment.)

Reply:
I've enabled the usual way to do cross-compilation, with a fix for the underlying issue I had here. Now, it fails at other places.